picamera2/picamera2.py

```diff
@@ -1061,7 +1126,7 @@ def process_requests(self) -> None:
             # immediately back to libcamera. You could consider customising this number.
             # When there's only one buffer in total, don't hang on to anything as it would stall
             # the pipeline completely.
-            max_len = 0 if self.camera_config['buffer_count'] == 1 else 1
+            max_len = 0 if (not self.camera_config or self.camera_config['buffer_count'] == 1) else 1
```

Comment:
Slightly curious about this one. Is it suggesting we've closed the camera but are still somehow in the event loop? Which perhaps suggests we failed to stop it, though maybe that could be related to the issues we had in the `stop_()` method. Maybe at some point I might undo this and see what breaks...!

Reply:
Good point, I'll try removing this after the latest changes.

Reply:
I've now reverted this change, after testing locally.